components/form-builder/app/navigation/LeftNavLink.tsx

```diff
@@ -1,10 +1,27 @@
+import { useTemplateContext } from "@components/form-builder/hooks";
+import { useTemplateStore } from "@components/form-builder/store";
 import Link from "next/link";
-import React, { ReactElement, useEffect, useState } from "react";
+import React, { ReactElement, useEffect, useState, useCallback } from "react";
 import { useActivePathname, cleanPath } from "../../hooks/useActivePathname";
 
 export const LeftNavLink = ({ children, href }: { children: ReactElement; href: string }) => {
   const [active, setActive] = useState(false);
   const { isReady, activePathname } = useActivePathname();
+  const { saveForm } = useTemplateContext();
+  const { setId, getIsPublished } = useTemplateStore((s) => ({
+    setId: s.setId,
+    getIsPublished: s.getIsPublished,
+  }));
+
+  const saveOnNavigate = useCallback(async () => {
+    if (!getIsPublished()) {
```

Comment:
Wondering if we should move more logic into saveForm

Potentially it could handle the `if` check for `getIsPublished` and also handling setting the id as needed.

Reply:
> allow to save only parts of the template that are allowed to be modified after publishing

nice ya 

>  Thinking that in the near future we would allow users to change delivery option if the form is already published.

👍